src/getter.js

```diff
@@ -102,6 +102,35 @@ function getStoreDeps(getter) {
   return storeDeps
 }
 
+/**
+ * Add override options to a getter
+ * @param {getter} getter
+ * @param {object} options
+ * @param {boolean} options.useCache
+ * @param {*} options.cacheKey
+ * @returns {getter}
+ */
+function addGetterOptions(getter, options) {
+  if (!isKeyPath(getter) && !isGetter(getter)) {
+    throw new Error('createGetter must be passed a keyPath or Getter')
+  }
+
+  if (getter.hasOwnProperty('__options')) {
```

Comment:
I think a `Getter` constructor alleviates some of this confusion, as it's clear that you are creating an instance of something and the options you provide with it would be tied to that instance.